### PR TITLE
Upgrade ESlint to v3.

### PR DESCRIPTION
**Pull request by jeresig**
Summary:
This upgrades ESlint to v3.

Two rules were disabled:
- Disabled react/sort-prop-types (after discussion in the Web Frontend meeting that this wasn't useful)
- Disabled no-case-declarations (this was causing problems with a common code pattern used in Redux-style code)

Test Plan:
I ran `./runlint.py ~/khan/webapp` and it generated warnings (as expected). I then fixed all the errors and running it again it passed!

Reviewers: kevinb, csilvers

### CODE REVIEW

**Comment by kevinbarabash** [comment]
LGTM.  Let's hold off on merging this until the lint fixes get deployed.

File: eslintrc.shared [hunk 1]
```diff
@@ -44,16 +45,15 @@
         "object-curly-spacing": 2,
         "one-var": [2, "never"],
         "semi": [2, "always"],
-        "space-after-keywords": 2,
```
Comment (csilvers) [inline eslintrc.shared]: These still seem useful things, did they go away in the new eslint?

File: eslintrc [hunk 2]
```diff
@@ -44,6 +43,8 @@
         // We turned this off since it was too much work for too
         // little benefit, especially for one-line props.
         "react/jsx-sort-props": 0,
+        // We turned this off too as we didn't see an explicit benefit
+        "react/sort-prop-types": 0,
```
Comment (csilvers) [inline eslintrc]: It looks like this used to be set but you cleared it as part of this review -- is that right?

Comment (jeresig) [inline eslintrc]: Correct!

**Comment by mroth** [comment]
Sidenote reminder in advance: As this is a decent sized change, please tag a semantic version release in GitHub once merged. Thanks!

**Comment by jeresig** [comment]
@mroth Will do, thank you!

**Pull request merged**